### ISSUE TITLE
Softmax poor performance

### ISSUE THREAD

**Issue by emartinezs44** [intel-analytics/analytics-zoo#5058]
The **com.intel.analytics.zoo.pipeline.api.keras.layers.SoftMax** layer shows a poor performance in comparison with other frameworks using the same hardware. In essence the operation from TensorNumeric _**vexp**_ . The tests that I have done with an input Tensor[Float] with dims as Array(1, 12, 512, 512) and Storage of 3145728 length it takes a mean of 0.65 seg per inference. As a result, models with attention that use this function in each layer to calculate the attention vector take so long. The Softmax layer in Torch is much faster in the same hardware. Any idea about this? how much faster is Mkl-dnn?

**Comment by yangw1234** [comment]
@emartinezs44 thank you for reporting this issue. We are looking into it.

**Comment by yangw1234** [comment]
@dding3 could you help take a look at this?

**Comment by jason-dai** [comment]
@emartinezs44 This should have been fixed by https://github.com/intel-analytics/analytics-zoo/pull/5076; you may verify the result in the latest nightly build.

**Comment by emartinezs44** [comment]
Perfect. I´ll check it out this evening. Thanks!

**Comment by emartinezs44** [comment]
My tests show that it is ~3.4 times faster. Good work!. I close the issue. Thanks.
